OTP authentication service (#1)
avantifellows/otp-service: README.md

Avanti's OTP service uses [Gupshup's](https://www.gupshup.io/developer/home) OTP API for authentication.

> This authentication is the dynamic generation of a numeric/alphanumeric/alphabetical code that authenticates the user for a single transaction or session. The code or one-time password (OTP) can be sent to the user via SMS.

Reviewer: ```suggestion
This service dynamically generates a numeric/alphanumeric/alphabetical code that authenticates the user for a single transaction or session. The code or one-time password (OTP) can be sent to the user via SMS.
```

> The API is protected, hence you need to add an API key to the header of each call.

Reviewer: ```suggestion
The API is protected, hence you need to add an API key to the header of each call. (See examples below)
```

> # Avanti Fellows OTP Service

Reviewer 2: use the github workflows and docs as added here: https://github.com/avantifellows/plio-aws-discord-alerts

This repository contains code for sending and verifying OTP sent to mobile numbers.

## Usage

> The service is exposed in the form of two endpoints

Reviewer 2: as you said in your description, this is really one endpoint with different functionalities based on the params, right? We should write that here.

Author: Our API has two endpoints, the Gupshup's API has only one.

### Send OTP

> Use the below endpoint to send OTP to a mobile number.

Reviewer 2: given that it is the same endpoint

```suggestion
Use the parameters below to send OTP to a mobile number.
```

```
Required query parameters: `phone`

> Optional query parameters: `msg`, `otpCodeLength`, `optCodeType`

Reviewer 3: Typo - `opt` instead of `otp`

Reviewer: whoops

}
response = requests.post(url, params = params)
```

### Verify OTP

> Use the below endpoint to verify OTP code sent to a mobile number.

Reviewer 2: same as before
```suggestion
Use the parameters below to verify OTP sent to a mobile number.
```

```

> https://ym6gqg7jui.execute-api.ap-south-1.amazonaws.com/dev/verifyotp

Reviewer 2: we should not expose the API link here. Instead, we can use it as an environment variable.
Then, you'll also need to add a section explaining what all the environment variables mean (currently, it would just be this one env variable)

Author: I'm confused about this. Are we not providing this OTP auth as a service? If we aren't, then why are we explaining how to use the API?

}
response = requests.post(url, params = params)
```

### Verify OTP
Use the below endpoint to verify OTP code sent to a mobile number.
```
https://ym6gqg7jui.execute-api.ap-south-1.amazonaws.com/dev/verifyotp
```
Required query parameters: `phone`, `code`

Optional query parameters: `msg`, `otpCodeLength`, `optCodeType`

#### Example
```py
import requests
url = "https://ym6gqg7jui.execute-api.ap-south-1.amazonaws.com/dev/verifyotp"
params = {
  "phone": "9999999999",
  "code":"1234"
}
response = requests.post(url, params = params)
```

### Query Parameters
`phone` : The number must be in pure numeric format with no special characters. <br/>

> `msg` : The message that needs to be sent. It can contain alphanumeric & special characters. The message must contain `%code%`.

Reviewer 3: This is currently not configurable, right? Or did you change this? If not configurable, mention it

Reviewer: It is configurable!
